SIL.Media/FFmpegRunner.cs

```diff
@@ -93,10 +108,45 @@ private static string LocateFFmpeg()
 			foreach (var path in progFileDirs)
 			{
 				var exePath = (Path.Combine(path, "FFmpeg for Audacity", kFFmpegExe));
-				if (File.Exists(exePath))
+				if (File.Exists(exePath) && MeetsMinimumVersionRequirement(exePath))
 					return exePath;
 			}
-			return null;
+
+			string ffmpeg = null;
+
+			// Locate may be called multiple times, we don't want to run this command every time.
+			if (_ffmpegOnPath == null)
+			{
+				ffmpeg = Path.GetFileNameWithoutExtension(kFFmpegExe);
+				// Try to just run ffmpeg from the path, if it works then we can use that directly.
+				_ffmpegOnPath = MeetsMinimumVersionRequirement(ffmpeg);
+				if (!_ffmpegOnPath.Value)
+					ffmpeg = null;
+			}
+			return ffmpeg;
+		}
+
+		private static bool MeetsMinimumVersionRequirement(string ffmpeg)
```

Comment:
are we concerned that this will execute every time (unless an explicit path is set) `LocateFFmpeg` is called? There's some places where it's called every time rather than using `LocateAndRememberFFmpeg`

Reply:
It looks like it has been that way for over a decade. Changing it would be a very subtle change. It's _potentially_ a breaking change but only in the really bizarre edge case where an app does not install FFmpeg and the user installs it while running the app. In the grand scheme of things re-locating it each time probably isn't a huge performance penalty, but it seems pretty clear that in general this class was set up to check once and then assume the situation would not change. Maybe these methods were ported over from some app that didn't have this concept. I think I'll fix it.